src/ceph-disk/ceph_disk/main.py

```diff
@@ -1519,6 +1519,36 @@ def zap(dev):
     if not stat.S_ISBLK(dmode) or is_partition(dev):
         raise Error('not full block device; cannot zap', dev)
     try:
+        # Thoroughly wipe all partitions from any traces of
+        # Filesystems or OSD Journals
+        #
+        # In addition we need to write 10M of data to each partition
+        # first to make sure that after re-creating the same parition
+        # there is no trace left of any previous Filesystem or OSD
+        # Journal
+
+        LOG.debug('Writing zeros to existing partitions on %s', dev)
+
+        for partname in list_partitions(dev):
+            partition = get_dev_path(partname)
+            command_check_call(
+                [
+                    'wipefs',
+                    '--all',
```

Comment:
Shouldn't we add `−−quiet` and `−−force` to this command? Force will be helpful if the fs is mounted. I haven't tried `wipefs` behaviour on a mounted partition...